app/editor/src/components/form/text/Text.tsx

```diff
@@ -30,23 +45,50 @@ export const Text: React.FC<ITextProps> = ({
   tooltip,
   children,
   className,
+  error,
+  width = FieldSize.Stretch,
+  onInput,
+  onInvalid,
   ...rest
 }) => {
+  const [errorMsg, setErrorMsg] = React.useState(error);
+
   return (
-    <div className="frm-in">
+    <styled.Text className="frm-in">
       {label && <label htmlFor={id ?? `txt-${name}`}>{label}</label>}
-      <styled.Text
+      <styled.TextField
         name={name}
         id={id}
         type={type}
         variant={variant}
         className={`txt ${className ?? ''}`}
         data-for="main-tooltip"
         data-tip={tooltip}
+        width={width}
+        role={errorMsg ? 'alert' : 'none'}
+        onInput={(e) => {
```

Comment:
This crazy `onInput` and `onInvalid` stuff needs to be added to form fields to handle native validation functionality.  I don't love it, but it appears to be working.